app/models/df_json_accessor.rb

```diff
@@ -0,0 +1,40 @@
+class DfJsonAccessor
+
+  def initialize(json)
+    @json = json
+  end
+
+  def self.define_json_accessors
```

Comment:
I don't 100% like the use of the word `accessor` because to me it implies an api parity with `attr_accessor` somewhere. It is internally consistent with our xml accessor though so I'll not block on it but I did want to say that explicitly.

Reply:
Oh, I mean in the sense that since:

```ruby
attr_accessor :foo, :bar, :baz
attr_reader :foo, :bar, :baz
attr_writer :foo, :bar, :baz
```

I'd expect to be able to

```ruby
json_accessor foo: { type: :string, key: 'some path' }
```

And the equivalent read/write APIs. But it's consistent with how we're doing xml stuff and I don't want to _block_ on it, necessarily.

Reply:
ah, gotcha! yes, that's _much_ cleaner & i've switched over to that syntax. thanks again for the extremely thoughtful review, and for getting into detail about your suggestions (loved the screenshot explaining the heredoc change - i've never used the <<-/<<~ syntax before so it helped me understand what you meant)!

Reply:
and of course, lmk if i was still not quite understanding what you were thinking

Reply:
in particular, i've seen you put multiple attribute names on a attr_reader/writer/accessor line before, but i wonder how you feel about that style now that each `json_reader` specification is a whole line unto itself. i switched from having multiple `json_reader`s to just one, but could switch back if you think it'd be clearer

Reply:
I may be inclined to break the hashes up over multiple lines but having them all be one call seems alright to me. Yea, this is exactly the sort of thing I was thinking about and looks good to me. Ultimately, either way you're having to set up a hash of keys -> configuration so may as well make declaring it and registering it a single step.